### PR TITLE
hyprland: 0.46.1 -> 0.46.2; hyprland-qtutils: init at 0.1.1

### PR DESCRIPTION
**Pull request by kolaente**
Looking for a fix for https://github.com/hyprwm/Hyprland/issues/8776 I've run the update script and included the updated version in my `configuration.nix`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by fufexan — APPROVED** [review]

**Review by ddogfoodd — APPROVED** [review]

**Comment by fufexan** [comment]
Sure. @kolaente do you mind creating the package in a new commit (nix file can be adapted from [upstream](https://github.com/hyprwm/hyprland-qtutils/blob/3833097e50473a152dd614d4b468886840b4ea78/nix/default.nix)) and adding me as a maintainer (along with yourself, if you want)? After that, add hyprland-qtutils to hyprland's `buildInputs`.

**Comment by kolaente** [comment]
@fufexan Added the package but I'm unsure about the missing build inputs. Can you check what's missing?

I've added a few and now it still fails with

```
CMake Error at /nix/store/4k2h9qsps52n1v56zq03yi67dwmxa81i-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:645 (message):
  The following required packages were not found:

   - wayland-protocols
   - cairo
   - pango
   - pangocairo
   - pixman-1
   - xcursor
   - libinput
   - gbm
   - re2
```

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (fufexan) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:16]: This doesn't need to be in a let. Can be left inside the `mkDerivation`.

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (fufexan) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:20]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

With this we can reference `finalAttrs.version` inside `fetchFromGitHub`.

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (fufexan) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:27]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

File: pkgs/top-level/all-packages.nix
Comment (fufexan) [inline pkgs/top-level/all-packages.nix:1]: Besides the typo, this is not needed at all. Packages in `pkgs/by-name` are automatically `callPackage`d.

**Review by fufexan — COMMENTED** [review]
Left a few comments. Not sure about the missing packages warning. Can you attach the whole log?

**Comment by jaredmontoya** [comment]
@kolaente In the hyprland flake hyprland-qtutils are added not to buildInputs or runtimeInputs but directly to path in postInstall
https://github.com/hyprwm/Hyprland/blob/main/nix/default.nix#L174

here is the same line in the current nixpkg
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/hy/hyprland/package.nix#L199

@fufexan I don't really understand what is the difference between runtimeInputs and adding things to path with a wrapper like in examples above. Is it better to do it the way hyprland flake does?

**Comment by fufexan** [comment]
I haven't seen `runtimeInputs` mentioned anywhere except [`writeShellApplication`](https://nixos.org/manual/nixpkgs/stable/#trivial-builder-writeShellApplication). A quick `rg` through Nixpkgs reveals its only usages are in that context.

I'm happy to be proven wrong, but I don't see special handling of it in stdenv.

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (khaneliman) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:27]:
```suggestion
    tag = "v${finalAttrs.version}";
```

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (khaneliman) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:45]:
```suggestion
    aquamarine
    hyprcursor
    hyprgraphics
    hyprlang
    hyprutils
    qt6.qtbase
    qt6.qtsvg
    qt6.qtwayland
```

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (khaneliman) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:54]:
```suggestion
    homepage = "https://github.com/hyprwm/hyprland-qtutils";
```

**Review by khaneliman — CHANGES_REQUESTED** [review]
Looks like you still need to add hyprland-qtutils to the buildInputs of hyprland.

**Review by JohnRTitor — DISMISSED** [review]
Looks like the tag attribute is not parsed properly.

@fufexan this started occuring after update to 0.46.0, could you take a look?
This is giving me warnings when a plugin for example pyprland fetches `hyprctl version -j`.

```bash

❯  nix shell github:nixos/nixpkgs/refs/pull/367113/merge#hyprland                      

  ~ 
❯  hyprctl version -j
{
    "branch": "",
    "commit": "71dc9f6128b8d2e382b2a574d2d5f15e2d907f3a",
    "version": "0.46.0",
    "dirty": false,
    "commit_message": "",
    "commit_date": "2024-12-19",
    "tag": "@TAG@",
    "commits": "5573",
    "buildAquamarine": "0.5.1",
    "buildHyprlang": "0.6.0",
    "buildHyprutils": "0.2.6",
    "buildHyprcursor": "0.1.10",
    "buildHyprgraphics": "0.1.1",
    "flags": ["debug"]
}
```

**Comment by fufexan** [comment]
@JohnRTitor you mean the @TAG@? This should not even be there, since we're now passing the appropriate tag from the info file.

**Comment by JohnRTitor** [comment]
I know maybe something broke from the source side.

**Comment by kolaente** [comment]
> Looks like you still need to add hyprland-qtutils to the buildInputs of hyprland.

yeah I thought getting the package to build first would be better.

**Comment by kolaente** [comment]
Thanks for all of your suggestions! I've added them all, but the build still fails with complaining about missing `gbm`. As far as I understood it, that is a protocol and `xboost` is an implementation of it. Adding `xboost` to the build inputs does not seem to work though.

<details><summary>Full error log</summary>

<pre>
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/dd837lga6pq6rrqaq8n8in4z1pah6via-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/share/doc/hyprland-qtutils -DCMAKE_INSTALL_INFODIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/ranlib -DCMAKE_AR=/nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/p591qpiacq7j5a82vw7l3m8sxzyfgdga-hyprland-qtutils-0.1.1
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Gathering git info
./scripts/generateVersion.sh: line 4: git: not found
./scripts/generateVersion.sh: line 5: git: not found
./scripts/generateVersion.sh: line 6: git: not found
./scripts/generateVersion.sh: line 7: git: not found
./scripts/generateVersion.sh: line 8: git: not found
./scripts/generateVersion.sh: line 9: git: not found
./scripts/generateVersion.sh: line 10: git: not found
-- Found PkgConfig: /nix/store/16h31npnbgpkhxp1041dsifiklbw4mpr-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'udis86>=1.7.2'
--   No package 'udis86' found
-- Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter)
Building itab.c/itab.h...
-- udis86 dependency not found, falling back to subproject
-- Found wayland-protocols at //nix/store/dxdbkrpc5gphjn0alqzkc8yvxbcy914v-wayland-protocols-1.38/share/wayland-protocols
-- Found wayland-scanner pkgdatadir at //nix/store/dx6m26y25y5z5f2mndinp7bp5vcjsb8w-wayland-scanner-1.23.1/share/wayland
-- Configuring Hyprland in Release with CMake
-- Checking deps...
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found OpenGL: /nix/store/ri85b3kn22dmr4yiimfsgd6r5x7zgk4z-libglvnd-1.7.0/lib/libOpenGL.so  found components: GLES3
-- Checking for module 'aquamarine>=0.4.5'
--   Found aquamarine, version 0.5.0
-- Checking for module 'hyprlang>=0.3.2'
--   Found hyprlang, version 0.6.0
-- Checking for module 'hyprcursor>=0.1.7'
--   Found hyprcursor, version 0.1.10
-- Checking for module 'hyprutils>=0.2.3'
--   Found hyprutils, version 0.2.6
-- Checking for module 'hyprgraphics>=0.1.1'
--   Found hyprgraphics, version 0.1.1
-- Checking for modules 'xkbcommon;uuid;wayland-server>=1.22.90;wayland-protocols;cairo;pango;pangocairo;pixman-1;xcursor;libdrm;libinput;gbm;gio-2.0;re2'
--   No package 'gbm' found
CMake Error at /nix/store/4k2h9qsps52n1v56zq03yi67dwmxa81i-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:645 (message):
  The following required packages were not found:

   - gbm

Call Stack (most recent call first):
  /nix/store/4k2h9qsps52n1v56zq03yi67dwmxa81i-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:873 (_pkg_check_modules_internal)
  CMakeLists.txt:116 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
</pre>
</details>

File: pkgs/by-name/hy/hyprland-qtutils/package.nix
Comment (khaneliman) [inline pkgs/by-name/hy/hyprland-qtutils/package.nix:51]:
```suggestion
    libgbm
    libinput
```

Do you just need to add this?

Comment (fufexan) [reply pkgs/by-name/hy/hyprland-qtutils/package.nix:51]: These dependencies are not even needed. Something went horribly wrong (I'm guessing hash reuse from Hyprland that confused Nix). I'll push a fix soon.

**Comment by fufexan** [comment]
@kolaente you seem to have disabled maintainer access to your fork, so here are the patches

<details>

<summary>qtutils fix</summary>

```diff
commit 47ad1ef537d44fab7bca52a468325655e0a2ae93
Author: Mihai Fufezan <mihai@fufexan.net>
Date:   Mon Dec 23 18:24:07 2024 +0200

    hyprland-qtutils: fix build

diff --git a/pkgs/by-name/hy/hyprland-qtutils/package.nix b/pkgs/by-name/hy/hyprland-qtutils/package.nix
index 64f674234e04..e9a6d8d29e6a 100644
--- a/pkgs/by-name/hy/hyprland-qtutils/package.nix
+++ b/pkgs/by-name/hy/hyprland-qtutils/package.nix
@@ -1,30 +1,17 @@
 {
   lib,
-  stdenv,
-  aquamarine,
-  cairo,
-  cmake,
+  gcc14Stdenv,
   fetchFromGitHub,
-  hyprcursor,
-  hyprgraphics,
-  hyprlang,
+  cmake,
+  pkg-config,
   hyprutils,
-  libinput,
-  pango,
   pciutils,
-  pixman,
-  pkg-config,
   qt6,
-  re2,
-  wayland-protocols,
-  xgboost,
-  xorg,
 }:
 let
-  inherit (lib.sources) cleanSource cleanSourceWith;
-  inherit (lib.strings) hasSuffix makeBinPath;
+  inherit (lib.strings) makeBinPath;
 in
-stdenv.mkDerivation (finalAttrs: {
+gcc14Stdenv.mkDerivation (finalAttrs: {
   pname = "hyprland-qtutils";
   version = "0.1.1";
 
@@ -32,7 +19,7 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "hyprwm";
     repo = "hyprland-qtutils";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-dj9dpVwpyTmUyVu4jtaIU39bHgVkoZjv6cgYfWyHc9E=";
+    hash = "sha256-Zp2cQOo1D7WH7ml5nV7mLyto0l19gTaWnbDb6zty1Qs=";
   };
 
   nativeBuildInputs = [
@@ -42,22 +29,10 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   buildInputs = [
-    aquamarine
-    cairo
-    hyprcursor
-    hyprgraphics
-    hyprlang
     hyprutils
-    libinput
-    pango
-    pixman
     qt6.qtbase
     qt6.qtsvg
     qt6.qtwayland
-    re2
-    wayland-protocols
-    xgboost
-    xorg.libXcursor
   ];
 
   preFixup = ''

```

</details>

<details>

<summary>Adding qtutils to hyprland</summary>

```diff
commit c88de3ba4f6c9bd0ceb55bd231cea9516c87f2f8
Author: Mihai Fufezan <mihai@fufexan.net>
Date:   Mon Dec 23 18:29:37 2024 +0200

    hyprland: add hyprland-qtutils to runtime

diff --git a/pkgs/by-name/hy/hyprland/package.nix b/pkgs/by-name/hy/hyprland/package.nix
index 303063b06493..8df4e6eebb39 100644
--- a/pkgs/by-name/hy/hyprland/package.nix
+++ b/pkgs/by-name/hy/hyprland/package.nix
@@ -15,6 +15,7 @@
   git,
   hyprcursor,
   hyprgraphics,
+  hyprland-qtutils,
   hyprlang,
   hyprutils,
   hyprwayland-scanner,
@@ -197,6 +198,7 @@ customStdenv.mkDerivation (finalAttrs: {
         --suffix PATH : ${
           makeBinPath [
             binutils
+            hyprland-qtutils
             pciutils
             pkgconf
           ]

```

</details>

**Comment by JohnRTitor** [comment]
> you seem to have disabled maintainer access to your fork,

Pushing to a contributor branch is only allowed from a user with write access afaik (nixpkgs-commiters)

**Comment by kolaente** [comment]
@fufexan applied the patches, please check again.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367113`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>hyprlandPlugins.hycov</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>hyprlandPlugins.hyprfocus</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 25 packages built:</summary>
  <ul>
    <li>grimblast</li>
    <li>hdrop</li>
    <li>hyprland</li>
    <li>hyprland-qtutils</li>
    <li>hyprland.dev</li>
    <li>hyprland.man</li>
    <li>hyprlandPlugins.borders-plus-plus</li>
    <li>hyprlandPlugins.csgo-vulkan-fix</li>
    <li>hyprlandPlugins.hy3</li>
    <li>hyprlandPlugins.hypr-dynamic-cursors</li>
    <li>hyprlandPlugins.hyprbars</li>
    <li>hyprlandPlugins.hyprexpo</li>
    <li>hyprlandPlugins.hyprgrass</li>
    <li>hyprlandPlugins.hyprscroller</li>
    <li>hyprlandPlugins.hyprspace</li>
    <li>hyprlandPlugins.hyprsplit</li>
    <li>hyprlandPlugins.hyprtrails</li>
    <li>hyprlandPlugins.hyprwinwrap</li>
    <li>hyprshade</li>
    <li>hyprshade.dist</li>
    <li>hyprshot</li>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>waybar</li>
    <li>xdg-desktop-portal-hyprland</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>hyprlandPlugins.hycov</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>hyprlandPlugins.hyprfocus</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 25 packages built:</summary>
  <ul>
    <li>grimblast</li>
    <li>hdrop</li>
    <li>hyprland</li>
    <li>hyprland-qtutils</li>
    <li>hyprland.dev</li>
    <li>hyprland.man</li>
    <li>hyprlandPlugins.borders-plus-plus</li>
    <li>hyprlandPlugins.csgo-vulkan-fix</li>
    <li>hyprlandPlugins.hy3</li>
    <li>hyprlandPlugins.hypr-dynamic-cursors</li>
    <li>hyprlandPlugins.hyprbars</li>
    <li>hyprlandPlugins.hyprexpo</li>
    <li>hyprlandPlugins.hyprgrass</li>
    <li>hyprlandPlugins.hyprscroller</li>
    <li>hyprlandPlugins.hyprspace</li>
    <li>hyprlandPlugins.hyprsplit</li>
    <li>hyprlandPlugins.hyprtrails</li>
    <li>hyprlandPlugins.hyprwinwrap</li>
    <li>hyprshade</li>
    <li>hyprshade.dist</li>
    <li>hyprshot</li>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>waybar</li>
    <li>xdg-desktop-portal-hyprland</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
LGTM

**Comment by fufexan** [comment]
I think the hyprland update and addition of qtutils to its path can be one commit.

**Comment by khaneliman** [comment]
> I think the hyprland update and addition of qtutils to its path can be one commit.

Ah, yes. Didn't see that. Have qtutils init first commit, then the bump including the new dependency.

**Review by JohnRTitor — COMMENTED** [review]
LGTM, will merge once commits are reordered

Btw anyone else able to reproduce this? https://github.com/NixOS/nixpkgs/pull/367113#pullrequestreview-2519913242

**Comment by fufexan** [comment]
@JohnRTitor I can reproduce that issue on the flake version as well. I'll see what I can find out about it, so please don't merge this in the meantime.

**Comment by khaneliman** [comment]
> @JohnRTitor I can reproduce that issue on the flake version as well. I'll see what I can find out about it, so please don't merge this in the meantime.

Is this issue introduced in 0.46.2? Or was it in 0.46.1?

**Comment by fufexan** [comment]
This is not a hyprland issue at all, it's a Nix string issue. `builtins.readFile ./VERSION` reads the `\n` as well, which breaks sed's invocation, as can be seen by prepending `set -x` to the `generateVersion.sh` script:

```bash
Running command: /nix/store/p6k7xp1lsfmbdd731mlglrdj2d66mr82-bash-5.2p37/bin/sh -c scripts/generateVersion.sh
--- stdout ---

--- stderr ---
+ cp -fr ./src/version.h.in ./src/version.h
+ HASH=
+ git branch --show-current
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
+ BRANCH=
+ git show
+ head -n 5
+ tail -n 1
+ sed -e 's/#//g' -e 's/\"//g'
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
+ MESSAGE=
+ DATE=2024-12-22
+ DIRTY=dirty
+ TAG='v0.46.0
'
+ COMMITS=
+ sed -i -e 's#@HASH@##' ./src/version.h
+ sed -i -e 's#@BRANCH@##' ./src/version.h
+ sed -i -e 's#@MESSAGE@##' ./src/version.h
+ sed -i -e 's#@DATE@#2024-12-22#' ./src/version.h
+ sed -i -e 's#@DIRTY@#dirty#' ./src/version.h
+ sed -i -e 's#@TAG@#v0.46.0
#' ./src/version.h
sed: -e expression #1, char 15: unterminated `s' command
+ sed -i -e 's#@COMMITS@##' ./src/version.h
```

This makes sense in the flake, since we do use `builtins.readFile`, but here it doesn't, since `info.tag` does not contain `\n`.

**Comment by kolaente** [comment]
I'm not on my computer right now, will reorder the commits later tonight. If someone with write permissions wants to reorder them before that, please go ahead.

**Comment by khaneliman** [comment]
@JohnRTitor Are you using the flake or this version from nixpkgs?

**Comment by JohnRTitor** [comment]
As you can see I got Hyprland (which includes hyprctl) in a `nix shell`, so this PR's Hyprland package should take precedence. 

> ```shell
> nix shell github:nixos/nixpkgs/refs/pull/367113/merge#hyprland                      
> 
> ```

**Comment by fufexan** [comment]
> As you can see I got Hyprland (which includes hyprctl) in a `nix shell`, so this PR's Hyprland package should take precedence.
> 
> > ```shell
> > nix shell github:nixos/nixpkgs/refs/pull/367113/merge#hyprland                      
> > ```

That's not how it works. Hyprctl, no matter where you got it from, will always get the version of the *currently running* Hyprland. Even if the one you shelled is newer. So I would recheck that.

Besides, running trying the same `set -x` in this PR yields this log

```bash
Running command: /nix/store/p6k7xp1lsfmbdd731mlglrdj2d66mr82-bash-5.2p37/bin/sh -c scripts/generateVersion.sh
--- stdout ---

--- stderr ---
+ cp -fr ./src/version.h.in ./src/version.h
+ HASH=0bd541f2fd902dbfa04c3ea2ccf679395e316887
+ BRANCH=v0.46.2-b
+ MESSAGE='version: bump to 0.46.2'
+ DATE=2024-12-19
+ DIRTY=
+ TAG=v0.46.2
+ COMMITS=0bd541f2fd902dbfa04c3ea2ccf679395e316887
+ sed -i -e 's#@HASH@#0bd541f2fd902dbfa04c3ea2ccf679395e316887#' ./src/version.h
+ sed -i -e 's#@BRANCH@#v0.46.2-b#' ./src/version.h
+ sed -i -e 's#@MESSAGE@#version: bump to 0.46.2#' ./src/version.h
+ sed -i -e 's#@DATE@#2024-12-19#' ./src/version.h
+ sed -i -e 's#@DIRTY@##' ./src/version.h
+ sed -i -e 's#@TAG@#v0.46.2#' ./src/version.h
+ sed -i -e 's#@COMMITS@#0bd541f2fd902dbfa04c3ea2ccf679395e316887#' ./src/version.h

```

and the resulting `src/version.h` file:

```c
#pragma once
#define GIT_COMMIT_HASH    "0bd541f2fd902dbfa04c3ea2ccf679395e316887"
#define GIT_BRANCH         "v0.46.2-b"
#define GIT_COMMIT_MESSAGE "version: bump to 0.46.2"
#define GIT_COMMIT_DATE    "2024-12-19"
#define GIT_DIRTY          ""
#define GIT_TAG            "v0.46.2"
#define GIT_COMMITS        "0bd541f2fd902dbfa04c3ea2ccf679395e316887"
```

**Comment by khaneliman** [comment]
@fufexan everything look good to you now?

**Review by fufexan — APPROVED** [review]
LGTM

**Comment by kolaente** [comment]
Thanks everyone!

**Comment by leiserfg** [comment]
@fufexan can/will this be backported to 24.11?

**Comment by fufexan** [comment]
Yes, will PR shortly.

**Comment by fufexan** [comment]
@leiserfg backport PR https://github.com/NixOS/nixpkgs/pull/369101

**Pull request merged**
